infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -1285,39 +1174,24 @@ private void deleteFilesMatchingFilter(FileFilter fileFilter) {
     @Override
     public void reset() {
         // Trim all segments
-        long endSegment = Math.max(logMetadata.getGlobalTail(), 0L) / RECORDS_PER_LOG_FILE;
+        long endSegment = getEndSegment();
```

Comment:
This is not robust. I know you didn't write it, but there is a problem.
If for some reason a node experiences data corruption, it is taken out of the cluster and then re-introduced. On the reset, we should do a "blind" delete and delete everything in /log , but this code depends on retrieving the end segment. what if that metadata is corrupted, then you won't be able to delete the state. 

A cleaner approach would be to just nuke everything in that directory.

Reply:
Yes, it's much simpler to delete everything, actually, I'm pretty sure in 100% cases we delete all the data anyway because `andSegment` should be always equal to the latest segment. 
Let me change this piece of code when we fix all the tests

Reply:
Everything in `*/corfu/log/*` and not everything in `/corfu/*` , still need to keep some of the local datastores data (i.e. managment/layoutserver layouts)